### DISCUSSION TITLE
Repetition of Loss value

### DISCUSSION THREAD

**Discussion by Aditi4AI** [mrdbourke/pytorch-deep-learning#527] [Q&A]
epoch=1
for epochs in range(epoch):
    model_0.train()
    Y_pred=model_0(X_train)
    loss=Loss(Y_pred,Y_train)
    Optimizer.zero_grad()
    loss.backward()
    print(f"Loss:{loss}")
    Optimizer.step()
    model_0.eval()
    with torch.inference_mode():
       Y_preds_new=model_0(X_test)
    plot_predictions(predictions=Y_preds_new);
   
I run this code for 32 times ,for the first 28 times  the output was descending i.e, going closer to zero :
Loss:0.19221457839012146--5th
Loss:0.1549803465604782--9th
Loss:0.08361945301294327--20th
Loss:0.035991813987493515--26th
but the after 30th time the values(Loss:0.03788195922970772) and (Loss:0.1076805368065834 )are repeating  no matter how many times I run the code these outputs are coming alternatively and are not decreasing further 
    with torch.inference_mode():
       Y_preds_new=model_0(X_test)
    plot_predictions(predictions=Y_preds_new);
    with torch.inference_mode():
        test_pred=model_0(X_test)
        test_loss=Loss(test_pred,Y_test)
    if epochs % 10 == 0:
        print (f"Epoch:{epochs}|Loss:{loss}|Test loss:{test_loss}")
        print (model_0.state_dict())
And also if did epoch=100 the outputs are
Epoch:10|Loss:0.03788195922970772|Test loss:0.1939408779144287
Epoch:90|Loss:0.03788195922970772|Test loss:0.1939408779144287
WHY?
